### DISCUSSION TITLE
v0.5 migration (aka, "hey, where's my stuff?")

### DISCUSSION THREAD

**Discussion by kjkent** [charmbracelet/soft-serve#403] [Q&A]
Hi devs and all,

After pulling the latest docker container after a while, I realised my repos were missing and, shortly after, that soft-serve has migrated away from the config repo. 

On the [v0.5 release notes](https://github.com/charmbracelet/soft-serve/releases/tag/v0.5.0), it's said to set a few environment variables and run `soft migrate-config` to bring things up to date. A few things that might help fellow Docker users:

- The release notes say to set three env vars, two of which are paths. Setting absolute paths do not work; `soft migrate-config` expects paths relative to `/soft-serve` (inside the container).

With this in mind, and at least in my case, `SOFT_SERVE_KEY_PATH="ssh/soft_serve_host_ed25519"` and `SOFT_SERVE_REPO_PATH="repos"`

- Running `soft migrate-config` as an ssh command just produces a "command not found" error. To run the command, do so at a bash prompt within the container, which you can access by running `docker exec -it <container-name> /bin/bash`. This works whether you run soft-serve by `docker run` or docker compose.

**Edit**: There looks to be some weirdness post-migration. After migrating, clearing the env vars in my `docker-compose.yaml`, and running the container, soft-serve logs a few lines of `Error: start server: create ssh server: ssh: no key found`.

I cleared the mapped volume folder, moving its contents to a second location, and reran the container, which started successfully using its defaults. I confirmed that the ssh `key_path` and `client_key_path` were identical (I originally had absolute paths in my YAML though these were persisted post-migration while other values were changed by the process. I changed these to match the default config to no avail):
```YAML
ssh:
  # ...
  key_path: "ssh/soft_serve_host_ed25519"
  client_key_path: "ssh/soft_serve_client_ed25519"
  # ...
```
And yet, I get the SSH key not found error when using the pre-migration key. Running the following fixed my issue in the context of using a compose file. I'd not recommend running this blindly:
```zsh
mv soft-serve soft-serve.bak
mkdir soft-serve
docker compose restart soft-serve
rm soft-serve/{soft-serve.db,config.yaml}
cp soft-serve.bak/{soft-serve.db,config.yaml} soft-serve/
docker compose restart soft-serve
```
Tl:dr, I backed up and wiped everything and let soft-serve recreate its /soft-serve folder, which is mapped locally. I then copied the config.yaml and soft-serve.db back, replacing their recreated versions, and now everything works.

**Comment by pspiagicw** [comment]
I had the same issue, there should be improved documentation on migration and a notice to existing users that their data might be gone.

**Comment by kjkent** [comment]
@aymanbagabas I just noticed you converted this issue into a discussion post and wonder why? It appears that the migrate command (at least back when this issue was posted) causes inaccessible servers due to the aforementioned SSH key error.

In any case, the migration experience led me to move to a more stable git server for the time being, but I keep an eye on anything Charm does and look forward to seeing how soft-serve develops!
